addons/stock/stock.py

```diff
@@ -503,8 +503,14 @@ def _quant_create(self, cr, uid, qty, move, lot_id=False, owner_id=False, src_pa
             'owner_id': owner_id,
             'package_id': dest_package_id,
         }
-
         if move.location_id.usage == 'internal':
+            if move.location_id.company_id.no_negative_stock:
+                raise Warning(_('You choose to avoid negative stock. You have \
+                    %s pieces of %s in location %s but you want to transfer \
+                    %s pieces. Please adjust your quantities or correct your \
+                    stock with an inventory adjustment if you have more \
+                    physically.') % (
```

Comment:
Please correct the error message :
You have chosen to avoid negative stock. %s pieces of %s are remaining in location %s but you want to transfer %s pieces. Please adjust your quantities or correct your stock with an inventory adjustment.